### PR TITLE
commands: redefine/refactor the new command interface

### PR DESCRIPTION
**Pull request by fanminshi**
cc/ @hasbro17

### CODE REVIEW

**Comment by fanminshi** [comment]
ignoring test result because controller-runtime is still in WIP.

File: commands/operator-sdk/cmd/new.go [hunk 1]
```diff
@@ -72,22 +62,17 @@ const (
...
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to create project %v: %v", projectName, err))
-	}
+	// TODO: codegen
 	pullDep()
 	initGit()
 }
 
 func parse(args []string) {
+	if len(args) != 1 {
+		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("new command needs 1 argument"))
```
Comment (hasbro17) [inline commands/operator-sdk/cmd/new.go]: Possibly stop using ExitWithError and simplify the above to:
```
logrus.Fatal("new command needs 1 argument")
```

Comment (fanminshi) [reply commands/operator-sdk/cmd/new.go]: yeah, let's do `log.Fatal()` instead of `logrus` because we don't really need the extra dependency.

**Comment by fanminshi** [comment]
all fixed. PTAL cc/ @hasbro17

**Comment by hasbro17** [comment]
LGTM

**Pull request merged**